app/models/ems_refresh.rb

```diff
@@ -195,6 +195,20 @@ def self.queue_merge(targets, ems, create_task = false)
       )
     end
 
+    if targets.present?
```

Comment:
looks like this always have all targets on a queue item (so the query grows huge), we will need to store only new_targets

on line 169
```
new_targets = targets
```

and here using new_targets

Reply:
please apply https://github.com/Ladas/manageiq/commit/30008f10630df012d065b2994af3df74a81e955b